src/rgb/standards.rs

```diff
@@ -0,0 +1,53 @@
+//!Various RGB standards.
+
+use num::Float;
+use Yxy;
+use flt;
+use white_point::{WhitePoint, D65};
+use rgb::{Primaries, RgbSpace, RgbStandard};
+use pixel::TransferFn;
+
+///The sRGB color space.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct Srgb;
+
+impl Primaries for Srgb {
```

Comment:
I have finally had some time to look into how the color calculators work, and I have found that easyrgb seems to ignore the white point when converting from RGB to XYZ/yxY and then uses it when converting to Lab, etc. So, it's `Srgb<D65> -> Xyz<?> -> Lab<Wp>`. I don't know if there's any adaption involved here, but it may very well be. The way I checked was the a tried to convert RGB(1, 0, 0) with D65 and D50 and saw that the XYZ and yxY values didn't change. I changed to RGB(1, 1, 1) and saw the same result. It was locked to a single white point.

Lindbloom's calculator is easier to inspect. The JavaScript source can be found [here](http://www.brucelindbloom.com/javascript/ColorConv.js) and the most interesting function is RGB2XYZ:

```javascript
function RGB2XYZ()
{
	var R = InvCompand(RGB.R);
	var G = InvCompand(RGB.G);
	var B = InvCompand(RGB.B);
	
	XYZ.X = R * MtxRGB2XYZ.m00 + G * MtxRGB2XYZ.m10 + B * MtxRGB2XYZ.m20;
	XYZ.Y = R * MtxRGB2XYZ.m01 + G * MtxRGB2XYZ.m11 + B * MtxRGB2XYZ.m21;
	XYZ.Z = R * MtxRGB2XYZ.m02 + G * MtxRGB2XYZ.m12 + B * MtxRGB2XYZ.m22;
	
	if (AdaptationMethod != 3)
	{
		var Ad = RefWhite.X * MtxAdaptMa.m00 + RefWhite.Y * MtxAdaptMa.m10 + RefWhite.Z * MtxAdaptMa.m20;
		var Bd = RefWhite.X * MtxAdaptMa.m01 + RefWhite.Y * MtxAdaptMa.m11 + RefWhite.Z * MtxAdaptMa.m21;
		var Cd = RefWhite.X * MtxAdaptMa.m02 + RefWhite.Y * MtxAdaptMa.m12 + RefWhite.Z * MtxAdaptMa.m22;
		
		var As = RefWhiteRGB.X * MtxAdaptMa.m00 + RefWhiteRGB.Y * MtxAdaptMa.m10 + RefWhiteRGB.Z * MtxAdaptMa.m20;
		var Bs = RefWhiteRGB.X * MtxAdaptMa.m01 + RefWhiteRGB.Y * MtxAdaptMa.m11 + RefWhiteRGB.Z * MtxAdaptMa.m21;
		var Cs = RefWhiteRGB.X * MtxAdaptMa.m02 + RefWhiteRGB.Y * MtxAdaptMa.m12 + RefWhiteRGB.Z * MtxAdaptMa.m22;
		
		var X = XYZ.X * MtxAdaptMa.m00 + XYZ.Y * MtxAdaptMa.m10 + XYZ.Z * MtxAdaptMa.m20;
		var Y = XYZ.X * MtxAdaptMa.m01 + XYZ.Y * MtxAdaptMa.m11 + XYZ.Z * MtxAdaptMa.m21;
		var Z = XYZ.X * MtxAdaptMa.m02 + XYZ.Y * MtxAdaptMa.m12 + XYZ.Z * MtxAdaptMa.m22;
		
		X *= (Ad / As);
		Y *= (Bd / Bs);
		Z *= (Cd / Cs);
		
		XYZ.X = X * MtxAdaptMaI.m00 + Y * MtxAdaptMaI.m10 + Z * MtxAdaptMaI.m20;
		XYZ.Y = X * MtxAdaptMaI.m01 + Y * MtxAdaptMaI.m11 + Z * MtxAdaptMaI.m21;
		XYZ.Z = X * MtxAdaptMaI.m02 + Y * MtxAdaptMaI.m12 + Z * MtxAdaptMaI.m22;
	}
}
```

It linealizes the RGB, converts it to XYZ, and _then_ performs chromatic adaption. This is essentially the same as what easyrgb does, but with a chromatic adaption step after conversion.

Lindbloom's calculator is closes to what we are trying to do, since it supports "custom" transfer functions. The way I understand it is that it would be the same as this, with the current implementation:

```rust
let xyz = Rgb::<(Srgb, TransferFn)>::new(r, g, b).into_xyz();
let final_xyz: Xyz<Wp> = xyz.adapt_into();
```

So, the question still remains: What's the meaning of `LinRgb<(Srgb, D50)>`? Do we force chromatic adaption of the primaries or not? I'm voting "no". I think we can still solve the ICC problem, but forcing adaption will close the door to inventing new RGB spaces by combining existing primaries with another white point.